src/cr/cube/dimension.py

```diff
@@ -530,8 +530,36 @@ def subtotals(self) -> "_Subtotals":
             insertion_dicts = self._dimension_transforms_dict["insertions"]
         # --- otherwise insertions defined on dimension/variable apply ---
         else:
-            view = self._dimension_dict.get("references", {}).get("view") or {}
-            insertion_dicts = view.get("transform", {}).get("insertions", [])
+            insertion_dicts = self._view_insertion_dicts
+        return _Subtotals(insertion_dicts, self.valid_elements)
+
+    @lazyproperty
+    def subtotals_with_payload_order(self) -> "_Subtotals":
+        """_Subtotals sequence object for this dimension respecting the payload order.
+
+        Each item in the sequence is a _Subtotal object specifying a subtotal, including
+        its addends and anchor.
+        """
+
+        def apply_order(insertion, view_insertions):
+            """
+            Change the anchor of the dimension transforms insertions with the
+            original paylod anchor.
+            """
+            name = insertion.get("name")
+            occurence = [el for el in view_insertions if el["name"] == name]
```

Comment:
Oh shoot, also this will have problems if they rename the subtotal. The frontend uses id, and if there's no id they use the index starting at 1.

Reply:
so on what we should rely on?

Reply:
Use id if present, and the 1 based index if not. I believe it’s always present on the analysis insertions, so you don’t have to worry about reordering affecting the ids